cip-0012.md

```diff
@@ -0,0 +1,80 @@
+<pre>
+  CIP: 12
+  Title: Memo Requirement through Broadcasts
+  Authors: Jeremy Johnson
+  Discussions-To: https://counterpartytalk.org/t/memo-requirement-through-broadcasts/3147
+  Status: Draft
+  Type: Standards
+  Created: 2017-06-07
+</pre>
+
+## Abstract ##
+
+Any user should be able to mark an address under their control as memo-required. The protocol should track this memo requirement and prevent any sends to the address without a memo.
+
+## Motivation ##
+
+To lessen end user support costs by enforcing a memo requirement at the protcol level.
+
+## Rationale ##
+
+CIP9 allows for specifying a memo in a send transaction, which can reduce the FEES cost for exchanges by allowing customers to make deposits directly to an exchange hot-wallet. The data in the 'memo' field is used to associate a transaction with a specific customer, and removes the requirement that each customer have their own deposit address. However, there is still a non-trivial SUPPORT cost associated with tracking down customer deposits _without_ a memo.
+
+By allowing an exchange to mark their address as memo-required, and rejecting sends without a memo, they are able to reduce their support costs.
+
+Enforcing a memo requirement at the protcol level also has the added benefit that it can discourage predatory behavior by exchanges, such as allowing sends without a memo, and ignoring any support requests by customers who made a deposit _without_ a memo. 
+
+## Specification ##
+
+There will be a 'special' broadcast message that is used to set address options; `OPTIONS <INTEGER>`.
+
+#### OPTIONS
+```OPTIONS <INTEGER>```
+ - `INTEGER`: a bitwise operation that identifes the desired options
+
+REQUIRE_MEMO is bit 1 set
+
+#### `LOCK`ed broadcast feed
+Counterparty has a feature to broadcast the text `LOCK` to prevent any further broadcasts from that address ever being considered valid.
+Any `OPTIONS` messages broadcast from an address with a `LOCK`ed broadcast feed will be ignored.
+
+## Changes ##
+
+### Database 
+ 
+1. Create `addresses` table with the following definition:
+```
+CREATE TABLE addresses (
+    address TEXT UNIQUE,    -- address
+    options INTEGER         -- bitwise op flag (1 << 0; 1 << 1)
```

Comment:
I'm not sure what `1 <<0; 1 << 1` means.  Perhaps change this to `-- bitwise address flags (MEMO_REQUIRED is options AND 0x01)`